core/domain/feedback_jobs_one_off.py

```diff
@@ -66,3 +67,34 @@ def map(item):
     @staticmethod
     def reduce(key, instance_ids):
         yield (key, instance_ids)
+
+
+class PopulateMessageCountOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for populating the message count field."""
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """Returns a list of datastore class references to map over."""
+        return [feedback_models.GeneralFeedbackThreadModel]
+
+    @staticmethod
+    def map(thread):
+        if thread.message_count in [None, 0]:
+            # Assigning the value of message_count if it is None.
+            thread.message_count = feedback_services.get_message_count(
+                thread.id)
+            try:
+                # Sets the message_count if it is None.
+                thread.put()
+                yield ('SUCCESS', thread.id)
+            except AttributeError:
+                yield ('FAILED', thread.id)
+        else:
+            yield ('NO-OP', thread.id)
+
+    @staticmethod
+    def reduce(message, thread_ids):
+        if message == 'FAILED':
+            yield (message, thread_ids)
+        else:
+            yield len(thread_ids)
```

Comment:
Yields in reduce() should always have a key and value.

Reply:
Will change it.

Reply:
DONE